Logic/Path/ball_path.py

```diff
@@ -55,14 +62,18 @@ def add_graph_edges(self):
 
                 if target_hit_position is not None:
                     if self.is_path_valid(self.white, target_hit_position, [self.white_index, target_index]):
-                        if self.is_possible_shot(self.white, target_ball_position, target_hole):                            
+                        if self.is_possible_shot(self.white, target_ball_position, target_hole):
                             distance = self.vectors.distance_from_two_points(self.white, target_hit_position)
                             self.graph.add_edge(self.white, target_hit_position, distance)
 
                             self.graph.add_edge(target_hit_position, target_ball_position, 0)
-
                             distance = self.vectors.distance_from_two_points(target_ball_position, target_hole)
                             self.graph.add_edge(target_ball_position, target_hole, distance)
+                        else:
+                            # Adding a constant distance to make such paths less favourtable than those that reach a hole
+
```

Comment:
Maybe clarify that this hits a random ball when no paths are available.

Reply:
Done